modules/neovim/init/basic.nix

```diff
@@ -106,63 +95,55 @@ in {
     # Set options that were previously interpolated in 'luaConfigRC.basic' as vim.options (vim.o)
     # and 'vim.globals' (vim.g). Future options, if possible, should be added here instead of the
     # luaConfigRC section below.
-    options = pushDownDefault {
-      encoding = "utf-8";
-      hidden = true;
-      expandtab = true;
-    };
-
-    globals = pushDownDefault {
-      editorconfig = cfg.enableEditorconfig;
-    };
-
-    # Options that are more difficult to set through 'vim.options'. Fear not, though
-    # as the Lua DAG is still as powerful as it could be.
+    options = pushDownDefault (lib.mergeAttrsList [
+      {
+        # Options that are always set, with a lower priority
+        encoding = "utf-8";
+        hidden = true;
+        expandtab = true;
+
+        # Junkfile Behaviour
+        swapfile = !cfg.preventJunkFiles;
+        backup = !cfg.preventJunkFiles;
+        writebackup = !cfg.preventJunkFiles;
+      }
+
+      (optionalAttrs cfg.undoFile.enable {
+        undofile = true;
+        undodir = cfg.undoFile.path;
+      })
+
+      (optionalAttrs (cfg.bell == "none") {
+        errorbells = false;
+        visualbell = false;
+      })
+
+      (optionalAttrs (cfg.bell == "on") {
+        visualbell = false;
+      })
+
+      (optionalAttrs (cfg.bell == "visual") {
+        visualbell = false;
+      })
+
+      (optionalAttrs (cfg.lineNumberMode == "relative") {
+        relativenumber = true;
+      })
+
+      (optionalAttrs (cfg.lineNumberMode == "number") {
+        number = true;
+      })
+
+      (optionalAttrs (cfg.lineNumberMode == "relNumber") {
+        number = true;
+        relativenumber = true;
+      })
```

Comment:
We should just deprecate these too, no point in abstractions

Reply:
I'm not keen on aggressive deprecations outside the 'release cycle.' We can consider fully deprecating those options in the 0.8 branchout.